pkg/event/events.go

```diff
@@ -40,12 +40,13 @@ type UserEvent struct {
 
 // ImpressionEvent represents an impression event
 type ImpressionEvent struct {
-	EntityID     string `json:"entity_id"`
-	Key          string `json:"key"`
 	Attributes   []VisitorAttribute
-	VariationID  string `json:"variation_id"`
-	CampaignID   string `json:"campaign_id"`
-	ExperimentID string `json:"experiment_id"`
+	CampaignID   string            `json:"campaign_id"`
+	EntityID     string            `json:"entity_id"`
+	ExperimentID string            `json:"experiment_id"`
+	Key          string            `json:"key"`
+	Metadata     map[string]string `json:"metadata"`
```

Comment:
`strictly` typed representation would be great here.

Reply:
:+1: I've replaced this with a `struct`